src/lib/operators/join_index.cpp

```diff
@@ -55,45 +62,55 @@ std::shared_ptr<const Table> JoinIndex::_on_execute() {
                   !_secondary_predicates.empty()),
          "JoinHash doesn't support these parameters");
 
-  _right_matches.resize(input_table_right()->chunk_count());
-  _left_matches.resize(input_table_left()->chunk_count());
+  std::shared_ptr<const Table> probe_input_table;
+  std::shared_ptr<const Table> index_input_table;
+
+  if (_index_side == JoinInputSide::Left) {
+    probe_input_table = input_table_right();
+    index_input_table = input_table_left();
+  } else {
+    probe_input_table = input_table_left();
+    index_input_table = input_table_right();
+  }
+
+  _index_matches.resize(index_input_table->chunk_count());
+  _probe_matches.resize(probe_input_table->chunk_count());
 
   const auto is_semi_or_anti_join =
       _mode == JoinMode::Semi || _mode == JoinMode::AntiNullAsFalse || _mode == JoinMode::AntiNullAsTrue;
 
-  const auto track_left_matches = _mode == JoinMode::FullOuter || _mode == JoinMode::Left || is_semi_or_anti_join;
-  const auto track_right_matches = _mode == JoinMode::FullOuter || _mode == JoinMode::Right;
+  const auto track_probe_matches = _mode == JoinMode::FullOuter || _mode == JoinMode::Left || is_semi_or_anti_join;
+  const auto track_index_matches = _mode == JoinMode::FullOuter || _mode == JoinMode::Right;
 
-  if (track_left_matches) {
-    for (ChunkID chunk_id_left = ChunkID{0}; chunk_id_left < input_table_left()->chunk_count(); ++chunk_id_left) {
-      _left_matches[chunk_id_left].resize(input_table_left()->get_chunk(chunk_id_left)->size());
+  if (track_probe_matches) {
+    for (ChunkID chunk_id_left = ChunkID{0}; chunk_id_left < probe_input_table->chunk_count(); ++chunk_id_left) {
+      _probe_matches[chunk_id_left].resize(probe_input_table->get_chunk(chunk_id_left)->size());
     }
   }
 
-  if (track_right_matches) {
-    for (ChunkID chunk_id_right = ChunkID{0}; chunk_id_right < input_table_right()->chunk_count(); ++chunk_id_right) {
-      _right_matches[chunk_id_right].resize(input_table_right()->get_chunk(chunk_id_right)->size());
+  if (track_index_matches) {
+    for (ChunkID chunk_id_right = ChunkID{0}; chunk_id_right < index_input_table->chunk_count(); ++chunk_id_right) {
```

Comment:
Still using "right" here. Please grep for left/right and make sure you catch them all.